### PR TITLE
davinci-resolve: fix desktop item, fix mainProgram for studio variant

### LINKED ISSUE #278133: DaVinci Resolve Studio is missing .desktop entry for application launchers
### Describe the bug
DaVinci Resolve Studio is missing a `.desktop` entry for application launchers

### Steps To Reproduce
1. `grep -Ri davinci /run/current-system/sw/share/applications/`

### Expected behavior
`/run/current-system/sw/share/applications/` should contain a `.desktop` entry for application launchers

### Notify maintainers
@jshcmpbll 
@amarshall

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.8, NixOS, 24.05 (Uakari), 24.05.20231224.5f64a12`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.19.2`
 - nixpkgs: `/nix/store/5acdh8xyry0kdvp6xla2hw7wf3zkphkl-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by amarshall**
## Description of changes

Should fix https://github.com/NixOS/nixpkgs/issues/278133. Unfortunately no icon file, perhaps it’s somewhere, but I just looked for “icon” files in the drv output and came up with only seemingly-irrelevant results.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/davinci-resolve/default.nix [hunk 1]
```diff
@@ -162,10 +162,10 @@ let
 
       desktopItems = [
         (makeDesktopItem {
-          name = "davinci-resolve";
-          desktopName = "Davinci Resolve";
+          name = "davinci-resolve${lib.optionalString studioVariant "-studio"}";
+          desktopName = "Davinci Resolve${lib.optionalString studioVariant " Studio"}";
           genericName = "Video Editor";
-          exec = "resolve";
+          exec = "davinci-resolve${lib.optionalString studioVariant "-studio"}";
```
Comment (pbsds) [inline pkgs/applications/video/davinci-resolve/default.nix:168]: could you also add this logic to `meta.mainProgram`?

**Review by jshcmpbll — APPROVED** [review]
The changes LGTM, and I have no objections to the inclusion of the .desktop file in the package. However, my experience with desktopItems is limited. Therefore, while I support these changes, I recommend that they be further reviewed by someone with more expertise in this area to confirm their correctness.

**Review by PowerUser64 — CHANGES_REQUESTED** [review]
Tried it out. Good news, the desktop item works! My only request is to add its icon.
image

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
It's still pending the requested change of setting `meta.mainProgram` correctly, and there is a merge conflict.

**Review by eclairevoyant — APPROVED** [review]

**Pull request merged**